src/Compilers/CSharp/Test/Emit2/Microsoft.CodeAnalysis.CSharp.Emit2.UnitTests.csproj

```diff
@@ -17,6 +17,12 @@
     <ProjectReference Include="..\..\Portable\Microsoft.CodeAnalysis.CSharp.csproj" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="..\Emit\Emit\EditAndContinue\EditAndContinueTestBase.cs">
+      <Link>Emit\EditAndContinue\EditAndContinueTestBase.cs</Link>
```

Comment:
><Link>Emit\EditAndContinue\EditAndContinueTestBase.cs</Link> [](http://example.com/codeflow?start=6&length=60)

I think it would be better to pull necessary helpers into an existing shared library. "Microsoft.CodeAnalysis.CSharp.Test.Utilities.csproj", for example. Alternatively, we can put affected tests into the test assembly with the helpers. As long as we don't hit the string heap size limit, we can do that.

Reply:
No problem, please check 1eed89b. Last time I tested the string limit, it was pretty close, any other PR merged before this is very likely to result this one to fail.

Reply:
Uh oh, turns out we have at least 4 `SemanticEditDescription` classes, pulling `EditAndContinueTestBase` to "Microsoft.CodeAnalysis.CSharp.Test.Utilities.csproj" cause EditorFeature test projects build fail. I think we have to stick to file linking for now.

Reply:
> Last time I tested the string limit, it was pretty close, any other PR merged before this one is very likely going to result this one to fail.

> I think we have to stick to file linking for now.

I think we just merged a PR that is going to add us some room in the Emit tests ([https://github.com/dotnet/roslyn/pull/58754](https://github.com/dotnet/roslyn/pull/58754)). Consider moving the affected tests into that test assembly instead.

Reply:
Great! Let me see.

Reply:
Moved, please check ffe1c99.